pkg/serviceimport/controller_test.go

```diff
@@ -118,26 +130,153 @@ func testLifecycleNotifications() {
+		})
+		It("should return the ip of the connected cluster everytime", func() {
+			_, _ = fakeClientset.LighthouseV2alpha1().ServiceImports(serviceImport.Namespace).Create(serviceImport)
+			si := newServiceImport(namespace1, service1, serviceIP2, clusterID2)
+			_, _ = fakeClientset.LighthouseV2alpha1().ServiceImports(si.Namespace).Create(si)
+
+			Eventually(func() bool {
+				first, _ := controller.serviceImports.SelectIP(namespace1, service1, mockCs.IsConnected)
+				second, _ := controller.serviceImports.SelectIP(namespace1, service1, mockCs.IsConnected)
+				return (first == serviceIP2) && (second == serviceIP2)
+			}).Should(BeTrue())
+		})
+	})
+
+	When("single service is present in 2 clusters with 2 ips in different namespace in the same cluster which is connected", func() {
+		JustBeforeEach(func() {
+			mockCs.clusterStatusMap[clusterID] = true
+		})
+		It("should return the same ip for each namespace", func() {
+			_, _ = fakeClientset.LighthouseV2alpha1().ServiceImports(serviceImport.Namespace).Create(serviceImport)
+			si := newServiceImport(namespace2, service1, serviceIP2, clusterID)
+			_, _ = fakeClientset.LighthouseV2alpha1().ServiceImports(si.Namespace).Create(si)
+
+			Eventually(func() bool {
+				first, _ := controller.serviceImports.SelectIP(namespace1, service1, mockCs.IsConnected)
+				second, _ := controller.serviceImports.SelectIP(namespace2, service1, mockCs.IsConnected)
+				third, _ := controller.serviceImports.SelectIP(namespace1, service1, mockCs.IsConnected)
+				forth, _ := controller.serviceImports.SelectIP(namespace2, service1, mockCs.IsConnected)
+				return (first == serviceIP) && (second == serviceIP2) && (third == serviceIP) && (forth == serviceIP2)
+			}).Should(BeTrue())
+		})
+	})
+
+	When("single service is present in 3 clusters with 3 ips in 3 clusters which are connected", func() {
+		JustBeforeEach(func() {
+			mockCs.clusterStatusMap[clusterID] = true
+			mockCs.clusterStatusMap[clusterID2] = true
+			mockCs.clusterStatusMap[clusterID3] = true
+		})
+		It("should return ip for each service in its turn according to rr", func() {
+			_, _ = fakeClientset.LighthouseV2alpha1().ServiceImports(serviceImport.Namespace).Create(serviceImport)
+			si1 := newServiceImport(namespace1, service1, serviceIP2, clusterID2)
+			_, _ = fakeClientset.LighthouseV2alpha1().ServiceImports(si1.Namespace).Create(si1)
+			si2 := newServiceImport(namespace1, service1, serviceIP3, clusterID3)
+			_, _ = fakeClientset.LighthouseV2alpha1().ServiceImports(si2.Namespace).Create(si2)
+
+			Eventually(func() bool {
+				first, _ := controller.serviceImports.SelectIP(namespace1, service1, mockCs.IsConnected)
+				second, _ := controller.serviceImports.SelectIP(namespace1, service1, mockCs.IsConnected)
+				third, _ := controller.serviceImports.SelectIP(namespace1, service1, mockCs.IsConnected)
+				return (first == serviceIP) && (second == serviceIP2) && (third == serviceIP3)
+			}).Should(BeTrue())
+		})
+	})
+}
```

Comment:
I think these tests should really be in a separate file specific to `serviceimport.Map`.  But we can leave it as is for now and I can push a patch afterwards so we can get this one in.